lib/adagrams.rb

```diff
@@ -0,0 +1,123 @@
+
+def draw_letters
+  all_letters = []
+  alphabet = {"A": 9, "B": 2, "C":2, "D": 4, "E": 12, "F": 2, "G": 3, "H": 2, "I": 9, "J": 1, "K": 1, "L": 4, "M": 2, "N": 6, "O": 8, "P": 2, "Q": 1, "R": 6, "S": 4, "T": 6, "U": 4, "V": 2, "W": 2, "X": 1, "Y": 2, "Z": 1}
+  alphabet.each do |letter, quantity|
+    letters = []
+    count = 0
+    while (count < quantity)
+      letters << letter.to_s
+      count += 1
+    end
+    all_letters << letters
+    all_letters = all_letters.flatten
+  end
+  user_hand = all_letters.sample(10)
+  return user_hand
+end
+
+
+#WAVE 2
+
+def uses_available_letters?(input, letters_in_hand)
+  result = true
+  input_array = input.upcase.chars
+  letters_in_hand2 = []
+  letters_in_hand2 += letters_in_hand
+  input_array.each do |char|
+    if !letters_in_hand2.include?(char)
+      result = false
```

Comment:
We haven't gone over this very well but I want to name that a method can have multiple return statements in it. 
This is kind of confusing because when any return statement is executed, the method immediately stops running but in a case like this, that's exactly what we want.
Because of that, we could just say `return false` here rather than setting result to false and waiting to return it until later.
This is considered standard practice because it means that (1) the program can finish without doing any extra work, once it's decided it wants to return false and (2) there's no longer a need for the result variable. (Any time we can do something with less code or less variables it's usually preferable.)